components/script/dom/mutationobserver.rs

```diff
@@ -21,16 +25,16 @@ pub struct MutationObserver {
 }
 
 impl MutationObserver {
-    fn new(global: &Window, callback: MutationCallback) -> MutationObserver {
+    fn new(global: &Window, callback: Rc<MutationCallback>) -> MutationObserver {
         MutationObserver {
             reflector_: Reflector::new(),
             callback: callback,
         }
     }
 
-    pub fn Constructor(global: &Window, callback: Rc<MutationCallback>) -> Result<Root<MutationObserver>, Error> {
-        let observer = MutationObserver::new(global, Rc::deref(&callback));
-        ScriptThread::add_mutation_observer(&observer)
+    pub fn Constructor(global: &Window, callback: Rc<MutationCallback>) -> Fallible<Root<MutationObserver>> {
+        let observer = MutationObserver::new(global, callback);
```

Comment:
You should be calling `reflect_dom_object` here ([per the documentation](http://doc.servo.org/script/dom/index.html#construction)); follow the pattern of other code that has a constructor method like [Blob](https://dxr.mozilla.org/servo/source/components/script/dom/blob.rs#85).